packages/frontend/src/retro/components/dialogs/CreateRetroSessionDialog.tsx

```diff
@@ -100,26 +99,28 @@ export function CreateRetroSessionDialog() {
       <DialogTitle id="form-dialog-create-retro">Create Retro Session</DialogTitle>
       <DialogContent sx={{ display: "flex", flexDirection: "column", gap: "1rem" }}>
         <Box>
-          <DialogContentText>Please enter your name</DialogContentText>
-          <TextInput
-            value={name}
+          <UserNameInputField
+            userName={name}
+            id="user-name"
             onSubmit={handleSubmit}
             onChange={handleNameChange}
             error={isNameError}
-            id="user-name"
-            label="Username"
-            autoFocus
+            isStorageAllowed={isStorageAllowed}
+            onStorageAllowanceChange={(event) => {
+              handleAllowanceChange(event.target.checked);
+            }}
           />
         </Box>
         <Box>
-          <DialogContentText>Please provide your name for this session</DialogContentText>
+          <DialogContentText>Please provide a name for this session</DialogContentText>
           <TextInput
             value={title}
             onSubmit={handleSubmit}
             onChange={handleTitleChange}
             error={isTitleError}
             id="retro-name"
             label="Retro Name"
+            autoFocus
```

Comment:
Ah, I see. You moved the autofocus to the session name. Is there a reason for this?